### ISSUE TITLE
API - List the muted hosts

### ISSUE THREAD

**Issue by iyappan-dcis** [Icinga/icinga2#6120]
I want to list the muted hosts (Notification disabled hosts using API ). I have read through this documentation and I don't find one. Please help.

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used (`icinga2 --version`):
* Operating System and version:
* Enabled features (`icinga2 feature list`):
* Icinga Web 2 version and modules (System - About):
* Config validation (`icinga2 daemon -C`):
* If you run multiple Icinga 2 instances, the `zones.conf` file (or `icinga2 object list --type Endpoint` and `icinga2 object list --type Zone`) from all affected nodes.

**Comment by Crunsher** [comment]
`curl -k -u root:icinga -H 'Accept: application/json' 'https://localhost:5665/v1/objects/hosts?enable_notifications=false'`

Does exactly that :)

**Comment by iyappan-dcis** [comment]
@Crunsher Thanks for your help. But this command does not filter the muted hosts instead it lists all the hosts in the below format.

{
            "attrs": {
                "__name": "ns-2.prod.he.fileright.com", 
                "acknowledgement": 0.0, 
                "acknowledgement_expiry": 0.0, 
                "action_url": "", 
                "active": true, 
                "address": "10.68.85.34", 
                "address6": "", 
                "check_attempt": 1.0, 
                "check_command": "ping", 
                "check_interval": 300.0, 
                "check_period": "", 
                "check_timeout": null, 
                "command_endpoint": "", 
                "display_name": "ns-2.prod.he.fileright.com", 
                "enable_active_checks": true, 
                "enable_event_handler": true, 
                "enable_flapping": false, 
                "enable_notifications": true, 
                "enable_passive_checks": true, 
                "enable_perfdata": true, 
                "event_command": "", 
                "flapping": false, 
                "flapping_last_change": 1519640074.266062, 
                "flapping_negative": 3829.0, 
                "flapping_positive": 0.0, 
                "flapping_threshold": 30.0, 
                "force_next_check": false, 
                "force_next_notification": false, 
                "groups": [
                    "FR"
                ], 
                "ha_mode": 0.0, 
                "icon_image": "", 
                "icon_image_alt": "", 
                "last_check": 1519640074.3324871, 
                "last_check_result": {
                    "active": true, 
                    "check_source": "icinga-3.prod.he.fileright.com", 
                    "command": [
                        "/usr/lib64/nagios/plugins/check_ping", 
                        "-H", 
                        "10.68.85.34", 
                        "-c", 
                        "200,15%", 
                        "-w", 
                        "100,5%"
                    ], 
                    "execution_end": 1519640074.33231, 
                    "execution_start": 1519640070.321198, 
                    "exit_status": 0.0, 
                    "output": "PING OK - Packet loss = 0%, RTA = 0.66 ms", 
                    "performance_data": [
                        "rta=0.656000ms;100.000000;200.000000;0.000000", 
                        "pl=0%;5;15;0"
                    ], 
                    "schedule_end": 1519640074.3324871, 
                    "schedule_start": 1519640370.3200002, 
                    "state": 0.0, 
                    "type": "CheckResult", 
                    "vars_after": {
                        "attempt": 1.0, 
                        "reachable": true, 
                        "state": 0.0, 
                        "state_type": 1.0
                    }, 
                    "vars_before": {
                        "attempt": 1.0, 
                        "reachable": true, 
                        "state": 0.0, 
                        "state_type": 1.0
                    }
                }, 
                "last_hard_state": 0.0, 
                "last_hard_state_change": 1517300423.0000939, 
                "last_in_downtime": false, 
                "last_reachable": true, 
                "last_state": 0.0, 
                "last_state_change": 1517300423.0000939, 
                "last_state_down": 1517300045.591727, 
                "last_state_type": 1.0, 
                "last_state_unreachable": 0.0, 
                "last_state_up": 1519640074.2660539, 
                "max_check_attempts": 3.0, 
                "name": "ns-2.prod.he.fileright.com", 
                "next_check": 1519640370.3199999, 
                "notes": "", 
                "notes_url": "", 
                "original_attributes": null, 
                "package": "_etc", 
                "paused": false, 
                "retry_interval": 60.0, 
                "state": 0.0, 
                "state_type": 1.0, 
                "templates": [
                    "ns-2.prod.he.fileright.com"
                ], 
                "type": "Host", 
                "vars": {
                    "dig_url": [
                        "www.google.com"
                    ], 
                    "notification": {
                        "mail": {
                            "groups": [
                                "icingaadmins"
                            ]
                        }
                    }, 
                    "ntp_address": "ns-1.prod.he.fileright.com", 
                    "ntp_critical": 2.0, 
                    "ntp_warning": 1.0, 
                    "os": "centos", 
                    "type": "FR"
                }, 
                "version": 0.0, 
                "volatile": false, 
                "zone": "icinga-3.prod.he.fileright.com"
            }, 
            "joins": {}, 
            "meta": {}, 
            "name": "ns-2.prod.he.fileright.com", 
            "type": "Host"
        },


Also will I be able to get the hostnames alone instead all these attributes ?

Thanks in Advance.

**Comment by Crunsher** [comment]
This is very weird, your host shows `"enable_notifications": true`!

As to filtering the output, you'll have to use some tool. Python works well for this. But maybe this is more a topic for https://monitoring-portal.org

**Comment by iyappan-dcis** [comment]
@Crunsher Below is the command that I used, Anything wrong with this command ?

curl -k -u icinga_user:icinga_pwd -H 'Accept: application/json' 'https://localhost:5665/v1/objects/hosts?enable_notifications=false'

**Comment by Crunsher** [comment]
Ah sorry, the path and query should be `v1/objects/hosts?filter=!host.enable_notifications` for disabled notifications and `v1/objects/hosts?filter=host.enable_notifications` for those with notifications enabled.

**Comment by iyappan-dcis** [comment]
@Crunsher That worked. Thanks alot!!

**Comment by iyappan-dcis** [comment]
@Crunsher I wanted to change the script to list the muted services alone and  I tried changing the script as below for the muted service notification and this did not work as expected. Please help

curl -k -u icinga_user:icinga_pwd -H 'Accept: application/json' 'https://localhost:5665/v1/objects/services?filter=!service.enable_notifications' | python -m json.tool

**Comment by iyappan-dcis** [comment]
@Crunsher Can you please provide me the api command for listing the muted services ? Please

**Comment by dnsmichi** [comment]
This is becoming a generic topic, please ask you API related questions on the community channels such as the forums or the mailing lists.
